trio/_abc.py

```diff
@@ -80,9 +80,11 @@ class Instrument(metaclass=ABCMeta):
 
     def before_run(self) -> None:
         """Called at the beginning of :func:`trio.run`."""
+        return
```

Comment:
lots of codecov misses, maybe `# pragma: no cover` on these methods?

Reply:
Or actually, maybe just add a test that calls all of these? The fact that inheriting from `Instrument` gives you do-nothing implementations of all the hook methods (so they won't error out) is something we probably want to guarantee.

Reply:
How would you guarantee that a function does nothing?

Reply:
We just want to test that calling it works - it exists and doesn't raise an exception.

Reply:
Added test in 9043b48